### PR TITLE
Pilot external testing

### PR DESCRIPTION
**Pull request by cpk1**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description

This is to address https://github.com/fastlane/fastlane/issues/6818
I've added the "betaState" attribute to Spaceship::Tunes:Build
In Pilot::BuildManager I've added the "betaState" status to the builds output.
Additionally I've added the parameter "external_available" to "distribute_build" this will be checked before calling "uploaded_build.build_train.update_testing_status!" 

### Motivation and Context

When building from jenkins and using a post-build action with "fastlane pilot upload distribute_external TRUE" pilot will always return an iTC error because it is impossible to turn on a build for external testing before beta app review, this in turn will cause the job to be marked as failed despite everything occurring as expected - build was uploaded and submitted for beta app review successfully.

https://github.com/fastlane/fastlane/issues/6818

I've uploaded and attempted to distribute for external testers at the same time (fastlane pilot upload distribute_external TRUE) and tried to distribute internally and externally builds that were already uploaded and approved and not approved for external distribution ("fastlane pilot distribute distribute_external TRUE" and "fastlane pilot distribute")

### CODE REVIEW

File: pilot/lib/pilot/build_manager.rb [hunk 1]
```diff
@@ -203,9 +215,17 @@ def distribute_build(uploaded_build, options)
...
       # Submit for beta testing
-      type = options[:distribute_external] ? 'external' : 'internal'
-      uploaded_build.build_train.update_testing_status!(true, type, uploaded_build)
-      return true
+      # We need to check if the user is attempting to turn on external builds and the status for external builds
+      # If we attempt to make an unapproved external build available iTC will return an error
+      if external_available || !options[:distribute_external]
+        type = options[:distribute_external] ? 'external' : 'internal'
+        uploaded_build.build_train.update_testing_status!(true, type, uploaded_build)
+        work_status = true
```
Comment (giginet) [inline pilot/lib/pilot/build_manager.rb]: I think you should return value here.

**Review by giginet — CHANGES_REQUESTED** [review]
Are there tests in this situation?

This logic seems to be complicated. So, you should add some specs.

**Comment by gcamp** [comment]
@cpk1 Are you planning to update your pull request? I would be available to fix it if you don't plan on doing it.

**Comment by cpk1** [comment]
I'm trying to take a look this week. Sorry for the accidental push!

**Comment by cpk1** [comment]
Oh I've noticed the change :). It looks like Pilot was affected quite a bit by todays update.

**Comment by KrauseFx** [comment]
@cpk1 that's ok, we're in the works on updating spaceship 👍

**Comment by ohayon** [comment]
I am going to close this because I think these changes have been superseded by the changes we needed to make when Apple rolled out changes to TestFlight a little while ago. thanks for this though @cpk1 !! 🚀

**Pull request closed without merging**